test/Z3/Base/Spec.hs

```diff
@@ -34,6 +34,60 @@ z3powerDef i j = i ^ j
 spec :: Spec
 spec = around withContext $ do
 
+
+  context "Sorts" $ do
+
+    specify "isEqSort" $ \ctx ->
+      monadicIO $ do
+        assert =<< (run $ do
+          s1 <- Z3.mkIntSort ctx
+          s2 <- Z3.mkIntSort ctx
+          Z3.isEqSort ctx s1 s2)
+        assert . not =<< (run $ do
+          s1 <- Z3.mkIntSort ctx
+          s2 <- Z3.mkRealSort ctx
+          Z3.isEqSort ctx s1 s2)
+        assert =<< (run $ do
+          s1 <- Z3.mkRealSort ctx
+          s2 <- Z3.mkRealSort ctx
+          Z3.isEqSort ctx s1 s2)
+        assert =<< (run $ do
+          s1 <- Z3.mkRealSort ctx
+          s2 <- Z3.mkRealSort ctx
+          s12 <- Z3.mkArraySort ctx s1 s2
+          s21 <- Z3.mkArraySort ctx s2 s1
+          Z3.isEqSort ctx s21 s12)
+        assert . not =<< (run $ do
+          s1 <- Z3.mkBoolSort ctx
+          s2 <- Z3.mkRealSort ctx
+          s12 <- Z3.mkArraySort ctx s1 s2
+          s21 <- Z3.mkArraySort ctx s2 s1
+          Z3.isEqSort ctx s21 s12)
+
+    specify "getSortId" $ \ctx -> 
+        sequence [ Z3.getSortId ctx =<< Z3.mkIntSort ctx
+                 , Z3.getSortId ctx =<< Z3.mkRealSort ctx
+                 , Z3.getSortId ctx =<< Z3.mkBoolSort ctx
+                 ] `shouldReturn`
+                 [ 2147483659
+                 , 2147483658
+                 , 2147483648
```

Comment:
NB: this is not a self-fulfilling test, I get the same IDs with the Python bindings.

Reply:
I'm not sure what is the typical use of `get_sort_id` but I wonder if these ids are an "internal detail" that could easily change. I think it would be better to write tests in the same way as for `isEqSort`. You could e.g. check that asking for the id of `Int` twice gives you the same result, and that the ids of `Int` and `Real` are different.

Reply:
I agree that it seems a bit hacky, but then again there is the API function to retrieve them. Your suggestion is good though, I change the test.